### PR TITLE
Cleanup IAutoComplete2

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Cleanup IAutoComplete2 interface
- Harden against invalid pointers
- Check HRESULTs rather than catch throwing PInvokes

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2031)

### CODE REVIEW

File: src/Common/src/Interop/Shell32/Interop.IAutoComplete2.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        [Guid("EAC04BC0-3791-11d2-BB95-0060977B464C")]
+        [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+        public unsafe interface IAutoComplete2
+        {
+            [PreserveSig]
+            HRESULT Init(
+                IntPtr hwndEdit,
+                IEnumString punkACL,
+                string pwszRegKeyPath,
+                string pwszQuickComplete);
```
Comment (weltkante) [inline src/Common/src/Interop/Shell32/Interop.IAutoComplete2.cs]: I think these need to be annotated as they are [`LPCWSTR`](https://docs.microsoft.com/en-us/windows/win32/api/shldisp/nf-shldisp-iautocomplete-init) but interop defaults to [`BSTR`](https://docs.microsoft.com/en-us/dotnet/framework/interop/blittable-and-non-blittable-types) if I read the docs right.

To your defense the entire Desktop declaration of all four methods had issues and you fixed 3 out of 4 :-)

Comment (hughbe) [reply src/Common/src/Interop/Shell32/Interop.IAutoComplete2.cs]: Good catch! Fixed

Comment (hughbe) [reply src/Common/src/Interop/Shell32/Interop.IAutoComplete2.cs]: But yes: you can pass a BSTR. For correctness this should be fixed!
> A BSTR is a pointer. The pointer points to the first character of the data string, not to the length prefix.

**Comment by RussKie** [comment]
@msftbot merge in 24 hours

**Pull request merged**